lib/ClientConnection.cc

```diff
@@ -254,6 +249,11 @@ ClientConnection::ClientConnection(const std::string& logicalAddress, const std:
 
         tlsSocket_ = ExecutorService::createTlsSocket(socket_, ctx);
 
+        if (!clientConfiguration.isTlsAllowInsecureConnection() && clientConfiguration.isValidateHostName()) {
+            LOG_DEBUG("Validating hostname for " << serviceUrl.host() << ":" << serviceUrl.port());
+            tlsSocket_->set_verify_callback(boost::asio::ssl::rfc2818_verification(serviceUrl.host()));
```

Comment:
When the client connects directly to the broker `serviceUrl` will be different from the actual broker hostname. If each broker has different certificates, this check would not be working

Reply:
The variable name is misleading, but ```serviceUrl``` is parsed ```physicalAddress```. So ```serviceUrl``` would be equal to actual broker hostname.
https://github.com/apache/pulsar-client-cpp/blob/v3.0.0/lib/ClientConnection.cc#L198-L199

Reply:
ping @merlimat PTAL again.

@izumo27 Please rebase to master to avoid broken CI.